### ISSUE TITLE
Warn about useless matches

### ISSUE THREAD

**Issue by grahamc** [rust-lang/rust-clippy#3664]
Clippy should warn about this useless match:

```rust
fn main()
{
    let a = 1;
    let b = 2;
    let c = 3;
    
    match (a, b, c) {
        (x, y, z) => {
            // this is a useless match
        }
    }
}
```

cc @oli-obk

**Comment by oli-obk** [comment]
This should not lint in macros, as it can frequently happen there and is not really an issue.

The suggestion should probably suggest `let (x, y, z) = (a, b, c);`. While we could do better, it seems very complicated to do so
